tools/check_projects_yaml.py

```diff
@@ -1,89 +1,138 @@
+                    Optional("latest_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("latest_release_version"): Or(float, str),
+                    Optional("first_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("first_release_version"): Or(float, str),
+                },  # type: ignore
+                # Emeritus GitHub projects
+                {
+                    **OPTIONAL,
+                    "name": str,
+                    "gh_url": check_url,
+                    "emeritus": True,
+                    Optional("url"): check_url,  # Overrides gh_url for the hyperlink
+                    Optional("release_count_zv"): int,
+                    Optional("first_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("first_release_version"): Or(float, str),
+                    Optional("first_nonzv_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("first_nonzv_release_version"): Or(float, str),
+                    Optional("last_zv_release_version"): Or(float, str),
+                },  # type: ignore
+                # Non-GitHub projects
+                {
+                    **OPTIONAL,
+                    "name": str,
+                    "url": check_url,
+                    Optional("emeritus"): False,
+                    Optional("release_count"): int,
+                    "first_release_date": Or(datetime.date, datetime.datetime),
+                    Optional("latest_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("latest_release_version"): Or(float, str),
+                    Optional("first_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("first_release_version"): Or(float, str),
+                },  # type: ignore
+                # Emeritus Non-GitHub projects
+                {
+                    **OPTIONAL,
+                    "name": str,
+                    "url": check_url,
+                    "emeritus": True,
+                    Optional("release_count_zv"): int,
+                    "first_release_date": Or(datetime.date, datetime.datetime),
+                    Optional("first_release_version"): Or(float, str),
+                    Optional("first_nonzv_release_date"): Or(
+                        datetime.date, datetime.datetime
+                    ),
+                    Optional("first_nonzv_release_version"): Or(float, str),
+                    Optional("last_zv_release_version"): Or(float, str),
+                },  # type: ignore
+            )
+        ],
+    },
+)
+
+
+def redundant(src: list[str]):
```

Comment:
Oh this one made its way into boltons, in case you want to import: https://boltons.readthedocs.io/en/latest/iterutils.html#boltons.iterutils.redundant